### PR TITLE
Add cluster notification

### PR DESCRIPTION
**Pull request by zhangqilike2015**

### CODE REVIEW

File: src/dashboard/server/api/validator/config.schema.json [hunk 1]
```diff
@@ -66,6 +66,10 @@
               "title": "Whether it supports allowed IP",
               "description": "Set to true if it could be listed in allowed ip page",
               "type": "boolean"
+            },
+            "notifications": {
+              "title": "Notifications for the cluster",
+              "type": "array"
```
Comment (Gerhut) [inline src/dashboard/server/api/validator/config.schema.json]: Shall we specify the type of the array's elements?

Comment (zhangqilike2015) [reply src/dashboard/server/api/validator/config.schema.json:72]: Good point, element should be string type, but how to implement, sorry for no idea.

Comment (zhangqilike2015) [reply src/dashboard/server/api/validator/config.schema.json]: > Shall we specify the type of the array's elements?

Good point, element should be string type, but how to implement, sorry for no idea.

Comment (Gerhut) [reply src/dashboard/server/api/validator/config.schema.json]: > > Shall we specify the type of the array's elements?
> 
> Good point, element should be string type, but how to implement, sorry for no idea.

https://json-schema.org/understanding-json-schema/reference/array.html#list-validation

Comment (zhangqilike2015) [reply src/dashboard/server/api/validator/config.schema.json]: > > > Shall we specify the type of the array's elements?
> > 
> > 
> > Good point, element should be string type, but how to implement, sorry for no idea.
> 
> https://json-schema.org/understanding-json-schema/reference/array.html#list-validation

thanks for help

File: src/dashboard/server/api/validator/config.schema.json [hunk 2]
```diff
@@ -66,6 +66,13 @@
               "title": "Whether it supports allowed IP",
               "description": "Set to true if it could be listed in allowed ip page",
               "type": "boolean"
+            },
+            "notifications": {
+              "title": "Notifications for the cluster",
+              "type": "array",
+                "items": {
```
Comment (Gerhut) [inline src/dashboard/server/api/validator/config.schema.json]: indent?

File: src/dashboard/src/Layout/NotificationBox.tsx [hunk 3]
```diff
@@ -30,29 +26,44 @@ const usePaperStyle = makeStyles(theme => createStyles({
...
+  const { clusters } = useContext(ClustersContext)
   return (
     <Box {...props}>
-      <Paper elevation={0} classes={paperStyle}>
-        <Info fontSize="small" color="primary"/>
-        <Typography variant="body2" component={Box} flex={1} paddingLeft={1}>{notification}</Typography>
-      </Paper>
-      <Divider/>
+      {clusters.map(cluster => (
+        <ClusterNotificationBox key={cluster.id} cluster={cluster}/>
```
Comment (Gerhut) [inline src/dashboard/src/Layout/NotificationBox.tsx]: Not a hard change request: I think `ClusterNotifications` is better then `ClusterNotificationBox`.

The `NotificationBox` is actually a `Box` (see line 63) but the `ClusterNotificationBox` is not.

File: src/dashboard/server/api/validator/config.schema.json (hunk 2 above)
Comment (zhangqilike2015) [reply src/dashboard/server/api/validator/config.schema.json]: > indent?

sorry for this typo

File: src/dashboard/src/Layout/NotificationBox.tsx (hunk 3 above)
Comment (zhangqilike2015) [reply src/dashboard/src/Layout/NotificationBox.tsx]: > Not a hard change request: I think `ClusterNotifications` is better then `ClusterNotificationBox`.
> 
> The `NotificationBox` is actually a `Box` (see line 63) but the `ClusterNotificationBox` is not.

I see, thanks. I need to be more formal.

**Review by Gerhut — APPROVED** [review]

**Pull request merged**